### ISSUE TITLE
Crash after upgrade to version 22

### ISSUE THREAD

**Issue by Zafaranec** [ReikaKalseki/Reika_Mods_Issues#2045]
Sorry, but I have unfortunately some problem, after update your mods to last version, I have same crash on client and server, here is paste: [crashlog](https://pastebin.com/rCqdgT64).

telling some incompatibility with Minefactory reloaded, but not have any problem with previous version.

**Comment by OvermindDL1** [comment]
I'm not seeing reika's mods appear anywhere in that stack trace.  Can you try it again without any of reika's mods and see if it crashes, if it doesn't then try it with just dragonapi, if that doesn't either then try it with just dragonapi and rotarycraft.

Also post the entire log preferably, at least the last few hundred lines, not just the crashlog as it misses contextual information like 'when' during the loading process it happens.

**Comment by Zafaranec** [comment]
English is not my native language and this will make my description not consistent, but I try it. In crashlog, you can see a list of mods in modpack, this modpack I make and continually updating about more than a year, with your mods version 21, is not any problem, and this version we play about month, when I try update your mods to version 22, keep crashing, same problem on client and on server, in crashlog is only some problem about minefactory reloaded, but we use this mod from start without problem, after I downgrade mods to version 21, no problem.

**Comment by OvermindDL1** [comment]
Yep, hence why the full logfile would be useful to see where in the loading process the crash happens, thus the full fml-client-latest.log file is needed, not just the crashlog.

**Comment by Zafaranec** [comment]
Oh, I'm sorry, I have log file and attach him with this post, not the latest log, because latest was successful but this is log with described problem. 
[fml-server-1.log](https://github.com/ReikaKalseki/Reika_Mods_Issues/files/2722380/fml-server-1.log)

**Comment by OvermindDL1** [comment]
Okay, so the PreInit event hits everything well it seems up until it tries to load `MineFactoryReloaded|CompatAtum`, at that point it sounds a lot like something is including MFR's API that shouldn't be including it...  Do you have a zip file of your entire mod set or can you grep through the jar's directly to see if anything is including an `IMFRHammer.class` file in any subdirectory?

**Comment by Zafaranec** [comment]
Sure, there is [link](https://www.dropbox.com/s/or9huem8r27secd/AoT_GT5u.zip?dl=1) to zip file, that we use for technicpack platform. I thanks a lot for your help, but my only question is, you change in update 22 about MFR or this class or...?

**Comment by OvermindDL1** [comment]
Update 22 added compat with the IMFRHammer interface in MFR again after not working as it's screwdriver for all of 1.7.10 until now, and I ran it myself with MFR with this same fix a few months ago without issue so I'm curious what else has changed, and this could be most likely caused by another mod holding the MFR API files (a big big No No that causes all kinds of issues) or another mod has caused a loading order issue or there is a loading issue somewhere or MFR's interface for it has changed since I messed with it (I ran the latest stable, not the latest beta, you are running the latest beta).

Downloading, extracting, it looks like that's the only instance of that class file, testing it's interface:
```zsh
╰─➤  javap IMFRHammer 
public interface powercrystals.minefactoryreloaded.api.IMFRHammer {
}
```
So it's still a simple marker class, no interface issue then.
MFR does not depend on anything reika at all.
So it has to be some other mod that is causing a loading dependency issue.

@ReikaKalseki 'might' be able to work around it by adding an optional dependency on MFR (whatever MFR's mod ID was, which it looks like it has not done in either RotaryCraft or DragonAPI on a cursory search).  This should be done anyway as an optional dependency should be on any mod that an interface is used from.

Excepting that, I'd say try running it with Just DragonAPI, RotaryCraft, and MFR as another test (I have no GUI right now or I'd try) and see if the same error happens.  If the same error does not happen than it's another mod causing the issue.

**Comment by OvermindDL1** [comment]
And for note, I'm not the mod author of these mods, I did other things.  ^.^

**Comment by ReikaKalseki** [comment]
There have been multiple reports of this error, so it is not a user-specific one. I was actually going to bring it up on the other issue.

**Comment by ReikaKalseki** [comment]
This is still being reported. If no solution is found soon I *will* revert to the old APIs.

**Comment by Skyler-Altol** [comment]
I have narrowed this crash down to being caused by ElectriCraft. Removing ElectriCraft from my dedicated server results in no crashes being observed on server startup.

**Comment by ReikaKalseki** [comment]
ElC does not use that API, and cannot be related.

**Comment by OvermindDL1** [comment]
> This is still being reported. If no solution is found soon I will revert to the old APIs.

The old API's were doing nothing since they don't exist in modern versions, so just commenting it out has the same effect.  ;-)

> I have narrowed this crash down to being caused by ElectriCraft. Removing ElectriCraft from my dedicated server results in no crashes being observed on server startup.

Yeah that's odd.  Does ElectriCraft use anything from within MFR at all?  Has it been tested with dragonapi having an optional dep on MFR?

**Comment by ReikaKalseki** [comment]
> Does ElectriCraft use anything from within MFR at all?

No.

>  Has it been tested with dragonapi having an optional dep on MFR?

I am surprised it does not already. What is its modID?

**Comment by OvermindDL1** [comment]
Hmm, looks like this:
```json
"modid": "MineFactoryReloaded",
```

**Comment by OvermindDL1** [comment]
And yeah, doesn't look like there is any dependency on it:
```java
╰─➤  grep -ER 'MineFactoryReloaded|MINEFACTORY' src/*
src/main/java/Reika/ReactorCraft/ReactorCraft.java:                     FMLInterModComms.sendMessage(ModList.MINEFACTORY.modLabel, "addLaserPreferredOre", tag);
src/main/java/Reika/DragonAPI/ModInteract/ItemHandlers/MFRHandler.java:         return ModList.MINEFACTORY;
src/main/java/Reika/DragonAPI/ModList.java:     MINEFACTORY("MineFactoryReloaded", "powercrystals.minefactoryreloaded.setup.MFRThings"),
src/main/java/Reika/DragonAPI/ModRegistry/ModWoodList.java:     MFRRUBBER(ModList.MINEFACTORY,  0x7E5C25, 0x5DC123, 30, 90, "rubberWoodBlock", "rubberLeavesBlock", "rubberSaplingBlock", new int[]{0,1,2,3,4,5,13}, new int[]{0,8}, 0, VarType.INSTANCE),
src/main/java/Reika/DragonAPI/DragonAPIInit.java:               this.registerHandler(ModList.MINEFACTORY, MFRHandler.class, "Handler");
src/main/java/Reika/ChromatiCraft/ModInterface/ModInteraction.java:                     ReflectiveFailureTracker.instance.logModReflectiveFailure(ModList.MINEFACTORY, e);
src/main/java/Reika/ChromatiCraft/ChromatiCraft.java:           if (ModList.MINEFACTORY.isLoaded()) {
src/main/java/Reika/VoidMonster/VoidMonster.java:               if (ModList.MINEFACTORY.isLoaded()) {
```
```java
        public static final String dependencies = MINFORGE+"after:BuildCraft|Energy;after:IC2;after:ThermalExpansion;after:Thaumcraft;"+
                        "after:powersuits;after:GalacticCraft;after:Mystcraft;after:UniversalElectricity;after:Forestry;after:MagicBees;"+
                        "after:ExtraBees;after:Natura;after:TConstruct;after:ProjRed|Core;after:bluepower;after:Waila;after:funkylocomotion;after:chisel;"+
                        "after:ComputerCraft;after:ThermalFoundation;after:CarpentersBlocks;after:AgriCraft";

```

**Comment by ProxyPlayerHD** [comment]
so we temporarly have to downgrade to V21? or wait for an update?

then again downgrading is out of the question as the only versions the mods have on Curse is V22 for some reason

**Comment by ReikaKalseki** [comment]
I have already added the above to the code for the next release.
